mr/src/test/java/org/apache/iceberg/mr/hive/TestTables.java

```diff
@@ -83,12 +92,107 @@ public Tables tables() {
-   * @param identifier The table identifier
-   * @return The location string for loadTable operation
+   * If creating the Hive table independently is needed for the given Catalog then this should return the Hive SQL
+   * string which is needed to be executed.
+   * @param identifier The table identifier (the namespace should be "default")
+   * @return The SQL string - which should be executed, null - if it is not needed.
+   */
+  public String createHiveTableSQL(TableIdentifier identifier) {
+    return String.format("CREATE TABLE default.%s STORED BY '%s' %s", identifier.name(),
+        HiveIcebergStorageHandler.class.getName(), locationForCreateTableSQL(identifier));
+  }
+
+  /**
+   * Loads the given table from the actual catalog. Overridden by HadoopTables, since the parameter of the
+   * {@link Tables#load(String)} should be the full path of the table metadata directory
+   * @param identifier The table we want to load
+   * @return The Table loaded from the Catalog
    */
-  public abstract String loadLocation(TableIdentifier identifier);
+  public Table loadTable(TableIdentifier identifier) {
+    return tables.load(identifier.toString());
+  }
+
+  /**
+   * Creates a Hive test table. Creates the Iceberg table/data and creates the corresponding Hive table as well when
+   * needed. The table will be in the 'default' database.
+   * @param shell The HiveShell used for Hive table creation
+   * @param tableName The name of the test table
+   * @param schema The schema used for the table creation
+   * @param fileFormat The file format used for writing the data
+   * @param records The records with which the table is populated
+   * @throws IOException If there is an error writing data
+   */
+  public void createTable(TestHiveShell shell, String tableName, Schema schema, FileFormat fileFormat,
+      List<Record> records) throws IOException {
+    createIcebergTable(shell.getHiveConf(), tableName, schema, fileFormat, records);
+    String createHiveSQL = createHiveTableSQL(TableIdentifier.of("default", tableName));
+    if (createHiveSQL != null) {
+      shell.executeStatement(createHiveSQL);
+    }
+  }
+
+  /**
+   * Creates a Hive test table. Creates the Iceberg table/data and creates the corresponding Hive table as well when
+   * needed. The table will be in the 'default' database.
+   * @param shell The HiveShell used for Hive table creation
+   * @param tableName The name of the test table
+   * @param schema The schema used for the table creation
+   * @param fileFormat The file format used for writing the data
+   * @param numRecords The number of records should be generated and stored in the table
+   * @param seed The seed used for the random record generation
+   * @throws IOException If there is an error writing data
+   */
+  public List<Record> createTableWithGeneratedRecords(TestHiveShell shell, String tableName, Schema schema,
+      FileFormat fileFormat, int numRecords, long seed) throws IOException {
```

Comment:
would it make sense to inline the seed variable since it's always 0 in the tests? Having the extra argument makes it harder to parse/eyeball the function calls in tests as well

Reply:
Done